src/odemis/driver/technolution.py

```diff
@@ -1746,6 +1732,24 @@ def _updateFrameDuration(self, _):
         field_scan_time = self.getTotalFieldScanTime()
         self.frameDuration._set_value(field_scan_time, force_write=True)
 
+    def updateMetadata(self, md):
+        if model.MD_FIELD_SIZE in md:
+            # The ASM API only accepts an effective field size that is a multiple of 32
+            eff_field_size = md.get(model.MD_FIELD_SIZE)
+            if eff_field_size[0] % 32 != 0 or eff_field_size[1] % 32 != 0:
+                # Calculate the suggested overlap only based on the first axis,
+                # because the overlap is a single number and it is just a suggestion.
+                if eff_field_size[0] % 32 <= 16:
+                    sug_field_size = eff_field_size[0] - eff_field_size[0] % 32
+                else:
+                    sug_field_size = eff_field_size[0] + 32 - eff_field_size[0] % 32
+                suggested_overlap = 1 - sug_field_size / self._scanner.resolution.value[0]
+                raise ValueError(
+                    f"Overlap must result in the value of the effective field size being a multiple of 32. "
+                    f"Effective field size is: {eff_field_size}, suggested overlap: {suggested_overlap * 100}%"
+                )
+        model.HwComponent.updateMetadata(self, md)
```

Comment:
Please use super() for calling the method of a parent class. It handles better complex class hierarchy, and is easier to copy-paste.
```suggestion
        super().updateMetadata(md)
```

Reply:
You could. I don't think it's very necessary though. For updateMetadata, I think it should be pretty safe. Be careful because in general that syntax is not necessary exactly identical. If the class inherits from a class which itself inherits from a parent class, it can also be a way to skip the parent class and directly call the grand-parent class. So not every case can be replaced by a `super()`.